pandas/core/missing.py

```diff
@@ -678,10 +660,15 @@ def clean_reindex_fill_method(method):
     return clean_fill_method(method, allow_nearest=True)
 
 
-def _interp_limit(invalid, fw_limit, bw_limit):
+def _interp_limit(
+    invalid: np.ndarray,
+    fw_limit: Optional[int],
+    bw_limit: Optional[int],
+    first: int,
```

Comment:
i'll update the algorithm so that passing first and last are unnecessary.

Reply:
https://github.com/pandas-dev/pandas/pull/34727/commits/1702ef50b63086b01e5b7aa4254d631d71b91113